### PR TITLE
Add clustering plots

### PR DESCRIPTION
**Pull request by daffidwilde**
I've added in the plots demonstrating the clustering methods, and a table describing the datasets.

I'd appreciate your opinion on how the visualisation looks @drvinceknight. To save you rendering it yourself, [here's a PDF](https://github.com/drvinceknight/literature-review/files/5674675/main.pdf)

### CODE REVIEW

**Comment by daffidwilde** [comment]
Also, I plan on archiving the source code used to make the datasets and the plots. I was thinking I would link the repo up to Zenodo, make a release and then include the DOI, but that feels a bit clumsy somehow. What would you recommend?

**Comment by drvinceknight** [comment]
> I was thinking I would link the repo up to Zenodo, make a release and then include the DOI, but that feels a bit clumsy somehow. What would you recommend?

I would just do a standalone archive of `img/clusters`: include a README in there and that's all you need. By standalone I mean directly to zenodo (so you don't go through GitHub -- happy to chat if I can clarify that).

**Comment by drvinceknight** [comment]
> I'd appreciate your opinion on how the visualisation looks @drvinceknight.

The graphs look absolutely gorgeous (the tree diagrams are fantastic).

I love how you **don't** have any labels on the scatter plots, it keep things clean and the labels are not the point **but** I wonder if this could be problematic from a reviewer point of view. Perhaps add something in the caption for each saying that the labels have been omitted as they're not relevant (and possibly even say what they are?). So something like:

`on the (a) moons (height versus width) (b) ellipses (foo versus bar) ...`

TLDR I think keeping the diagrams as clean as you have is wonderful but I can imagine a reviewer just going in to auto mode and finding it problematic so let us overload the captions.

File: sec/review.tex [hunk 1]
```diff
@@ -246,6 +247,55 @@ \subsubsection{Centroid-based clustering}\label{subsubsec:kmeans}
...
+    \caption{A summary of the synthetic datasets}\label{tab:datasets}
+\end{table}
+
+Figure~\ref{fig:kmeans_examples} demonstrates the nuances of centroid-based
+clustering discussed in this section, and shows how the \(k\)-means algorithm
+performs on three synthetic datasets. A brief summary of these datasets is
+provided in Table~\ref{tab:datasets}. The datasets were created using the Python
+library Scikit-learn~\cite{scikit-learn}, and the source code to generate them
+(and the plots included in this survey) has been archived online
+under~\doi{doi-info}.  % TODO archive online
```
Comment (drvinceknight) [inline sec/review.tex]:
```suggestion
library Scikit-learn~\cite{scikit-learn}, and the source code to generate them
(and the plots included in this survey) is available at the GitHub repository for 
this manscript \url{}  % TODO Add GitHub repo
and has also been archived online
under~\doi{doi-info}.  % TODO archive online
```

File: sec/review.tex [hunk 2]
```diff
@@ -246,6 +247,55 @@ \subsubsection{Centroid-based clustering}\label{subsubsec:kmeans}
 agnostic to its distance measure, and removes the requirement for the attribute
 space to be a metric space at all.
 
+\begin{table}
+    \input{tex/datasets.tex}
```
Comment (drvinceknight) [inline sec/review.tex:252]: :le_chefs_kiss:

**Review by drvinceknight — CHANGES_REQUESTED** [review]

**Comment by daffidwilde** [comment]
> I would just do a standalone archive of `img/clusters`

@drvinceknight so just archive the plots themselves, and leave the source code hosted on GitHub. Then, the `README` points at the repo?

File: sec/review.tex (hunk 2 above)
Comment (daffidwilde) [reply sec/review.tex:252]: I was sitting there thinking "how exactly does `pandas` make these tables look so nice?" 🤔

And then it dawned on me: just make `pandas` make the table look nice 🧑‍🍳🤌

File: sec/review.tex (hunk 1 above)
Comment (daffidwilde) [reply sec/review.tex]: Cool -- I didn't see this before I commented.

To reiterate [my comment](https://github.com/drvinceknight/literature-review/pull/1#issuecomment-743097530) below: you would leave the source code (a Jupyter notebook) on GitHub, and only archive the plots along with a useful `README`?

**Comment by daffidwilde** [comment]
> > I'd appreciate your opinion on how the visualisation looks @drvinceknight.
> 
> The graphs look absolutely gorgeous (the tree diagrams are fantastic).
> 
> I love how you **don't** have any labels on the scatter plots, it keep things clean and the labels are not the point **but** I wonder if this could be problematic from a reviewer point of view. Perhaps add something in the caption for each saying that the labels have been omitted as they're not relevant (and possibly even say what they are?). So something like:
> 
> `on the (a) moons (height versus width) (b) ellipses (foo versus bar) ...`
> 
> TLDR I think keeping the diagrams as clean as you have is wonderful but I can imagine a reviewer just going in to auto mode and finding it problematic so let us overload the captions.

As the labels are arbitrary, I'd prefer to just add a caveat in the main text highlighting that the labels have been removed because they don't matter.

**Comment by drvinceknight** [comment]
> @drvinceknight so just archive the plots themselves, and leave the source code hosted on GitHub. Then, the `README` points at the repo?

No I'd put the source code in the archive as well. I assumed all the code and images was in that `img/clusters` directory.

What I'd do is modify the repo so that it looks like:

```
img/clusters
    | --- k-means  # or the name of the figure
            |--- REAMDE.md
            |--- main.py
            | --- main.pdf
            | --- main.csv   # if relevant
    | --- <Tree thing> 
etc
```

Then you only archive the `clusters` dir.

**Comment by drvinceknight** [comment]
> As the labels are arbitrary, I'd prefer to just add a caveat in the main text highlighting that the labels have been removed because they don't matter.

I'm fine with that, it could be low hanging fruit for a reviewer but we can cross that bridge when we get to it.

**Comment by daffidwilde** [comment]
@drvinceknight I think this is ready to be merged. Let me know if there's anything else to add.

**Comment by drvinceknight** [comment]
> @drvinceknight I think this is ready to be merged. Let me know if there's anything else to add.

Yup this looks good to me :+1:

**Review by drvinceknight — APPROVED** [review]

**Comment by daffidwilde** [comment]
@drvinceknight great stuff. Can you merge it please? (I don't have write access on your fork)

**Comment by drvinceknight** [comment]
> Can you merge it please? (I don't have write access on your fork)

Duh 🤦
